src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -79,24 +79,38 @@ export default class AdyenV2PaymentStrategy implements PaymentStrategy {
                     throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
                 }
 
-                if (paymentData && isVaultedInstrument(paymentData) && isCardState(componentState)) {
-                    const { encryptedCardNumber, encryptedSecurityCode, encryptedExpiryMonth, encryptedExpiryYear } = componentState.data.paymentMethod;
-
-                    return this._store.dispatch(this._paymentActionCreator.submitPayment({
-                        ...payment,
-                        paymentData: {
-                            formattedPayload: {
-                                bigpay_token: {
-                                    token: paymentData.instrumentId,
-                                    credit_card_number_confirmation: encryptedCardNumber,
-                                    expiry_month: encryptedExpiryMonth,
-                                    expiry_year: encryptedExpiryYear,
-                                    verification_value: encryptedSecurityCode,
+                if (paymentData && isVaultedInstrument(paymentData)) {
+                    if (isCardState(componentState)) {
+                        const { encryptedCardNumber, encryptedSecurityCode, encryptedExpiryMonth, encryptedExpiryYear } = componentState.data.paymentMethod;
+
+                        return this._store.dispatch(this._paymentActionCreator.submitPayment({
```

Comment:
most of the logic seems to be repeated in both flows, maybe you could instead assemble the `bigpay_token ` according to the `if` condition and leave the rest outside?
e.g.

```js
if (isAccountState(componentState)) {
  bigpay_token = {
    ...bigpayToken,
    credit_card_number_confirmation: encryptedCardNumber,
    // ...
}

this._store.dispatch(this._paymentActionCreator.submitPayment({
  // ...
});